[system test] [doc] kafka package (#10650)
strimzi/strimzi-kafka-operator: development-docs/systemtests/io.strimzi.systemtest.kafka.TieredStorageST.md

# TieredStorageST

**Description:** This test suite covers scenarios for Tiered Storage integration implemented within Strimzi.

**Before tests execution steps:**

| Step | Action | Result |
| - | - | - |
| 1. | Create test namespace | Namespace is created |
| 2. | Build Kafka image based on passed parameters like image full name, base image, Dockerfile path (via Kaniko or OpenShift build) | Kafka image is built |
| 3. | Deploy Minio in test namespace and init the client inside the Minio pod | Minio is deployed and client is initialized |
| 4. | Init bucket in Minio for purposes of these tests | Bucket is initialized in Minio |

> | 5. | Deploy Strimzi Cluster Operator | Strimzi Cluster Operator is deployed |

Reviewer: ```suggestion
| 5. | Deploy Cluster Operator | Cluster Operator is deployed |
```

**Labels:**

* [kafka](labels/kafka.md)

<hr style="border:1px solid">

## testTieredStorageWithAivenPlugin

> **Description:** This testcase is focused on testing of Tiered Storage integration implemented within Strimzi. The tests use Aiven Tiered Storage plugin - <a href="https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/tree/main">...</a>

Reviewer: ```suggestion
**Description:** This testcase is focused on testing of Tiered Storage integration implemented within Strimzi. The tests use Aiven Tiered Storage plugin (<a href="https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/tree/main">tiered-storage-for-apache-kafka</a>).
```

**Steps:**

| Step | Action | Result |
| - | - | - |

> | 1. | Deploys KafkaNodePool resource with Broker NodePool with PV of size 10Gi | KafkaNodePool resource is deployed successfully with specified configuration |

Reviewer: ```suggestion
| 1. | Deploys KafkaNodePool resource with PV of size 10Gi | KafkaNodePool resource is deployed successfully with specified configuration |
```

| 2. | Deploys Kafka resource with configuration of Tiered Storage for Aiven plugin, pointing to Minio S3, and with image built in beforeAll | Kafka resource is deployed successfully with Tiered Storage configuration |

> | 3. | Creates topic with enabled Tiered Storage sync with size of segments set to 10mb (this is needed for speedup the sync) | Topic is created successfully with Tiered Storage enabled and segment size of 10mb |

Reviewer: ```suggestion
| 3. | Creates topic with enabled Tiered Storage sync with size of segments set to 10mb (this is needed to speed up the sync) | Topic is created successfully with Tiered Storage enabled and segment size of 10mb |
```
